### PR TITLE
feat: add fieldManager when patching Kubernetes resource

### LINKED ISSUE #6186: Declare a fieldManager name when Apply'ing a Kubernetes resource
### Is your enhancement related to a problem? Please describe

When editing a Kubernetes resource from the Kube tab, an apply command is done, and the command is visible in the `metadata.managedFields` field of the resource, but the manager is `unknown`. 

```
  managedFields:
    - apiVersion: apps/v1
      fieldsType: FieldsV1
      fieldsV1:
        [...]
      manager: kubectl-create    <-- done by 'kubectl create deployment ...'
      operation: Update
      time: 2024-02-28T13:47:46.000Z
    - apiVersion: apps/v1
      fieldsType: FieldsV1
      fieldsV1:
        [...]
      manager: unknown           <-- done by podman desktop
      operation: Update
      time: 2024-02-28T14:17:56.000Z

```

### Describe the solution you'd like

A value for manager specific to podman desktop, for example 'podman-desktop'

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Declare a field manager name when doing PATCH operation on a Kubernetes resource.

### Screenshot / video of UI

In the screenshot:
- the first Update is done by a `kubectl create deployment ...` command (manager = kubectl-create)
- the second is made by Podman Desktop before this PR (manager = unknown)
- the third is made by Podman Desktop with this PR (manager = podman-desktop)

fieldManager

### What issues does this PR fix or reference?

Fixes #6186 

### How to test this PR?

Create a deployment, then open this deployment in Podman Desktop and edit it from the Kube tab. 

In the resulting yaml, the `managedFields` section should indicate 'podman-desktop' as menager for the matching entry.

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
AFAIK it's missing a test

**Review by deboer-tim — APPROVED** [review]
Ditto, approved but needs test.

**Review by jeffmaury — CHANGES_REQUESTED** [review]
Test missing

**Review by cdrage — APPROVED** [review]
I see a test has been added, confirmed functional with a test YAML.

LGTM

File: packages/main/src/plugin/kubernetes-client.spec.ts [hunk 1]
```diff
@@ -1288,15 +1288,32 @@ test('Expect apply should patch if object exists', async () => {
...
   expect(objects).toHaveLength(1);
   expect(objects[0]).toEqual(patchedObj);
+  expect(patchMock).toHaveBeenCalledWith(expect.any(Object), undefined, undefined, 'podman-desktop');
+});
+
+test('Expect apply should patch with specific field manager', async () => {
+  const client = createTestClient('default');
+  const manifests = { kind: test, metadata: { annotations: test } } as unknown as KubernetesObject;
+  const patchedObj = { kind: 'patched' };
+  vi.spyOn(client, 'loadManifestsFromFile').mockReturnValue(Promise.resolve([manifests]));
```
Comment (benoitf) [inline packages/main/src/plugin/kubernetes-client.spec.ts]: I suppose it should be mockResolvedValue([manifests]) as it's an async method
```suggestion
  vi.spyOn(client, 'loadManifestsFromFile').mockResolvedValue([manifests]);
```

**Review by benoitf — APPROVED** [review]

File: packages/main/src/plugin/kubernetes-client.spec.ts (hunk 1 above)
Comment (feloy) [reply packages/main/src/plugin/kubernetes-client.spec.ts]: The pattern appears several times in the file, I'll update all of them

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**